### ISSUE TITLE
cannot parse CHECK constraint

### ISSUE THREAD

**Issue by tlgs** [sqlfluff/sqlfluff#1450]
I'm writing this bug report hoping `CREATE TABLE` / `ALTER TABLE` linting falls within the scope of this project.

## Expected Behaviour

Parse and lint `CREATE TABLE` statements containing `CHECK` constraints.

## Observed Behaviour

`PRS` error on files containing the `CHECK` constraint.

## Steps to Reproduce

```sql
CREATE TABLE users (
    username TEXT,
    age INT CHECK(age > 18)
);
```

```console
$ sqlfluff lint test.sql 
== [test.sql] FAIL
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'CREATE
                       | TABLE users (\n username TEXT,\n...'
WARNING: Parsing errors found and dialect is set to 'ansi'. Have you configured your dialect?
All Finished 📜 🎉!
```

```console
$ sqlfluff parse test.sql 
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        unparsable:                                           !! Expected: 'StatementSegment'
[L:  1, P:  1]      |            raw:                                              'CREATE'
[L:  1, P:  7]      |            whitespace:                                       ' '
[L:  1, P:  8]      |            raw:                                              'TABLE'
[L:  1, P: 13]      |            whitespace:                                       ' '
[L:  1, P: 14]      |            raw:                                              'users'
[L:  1, P: 19]      |            whitespace:                                       ' '
[L:  1, P: 20]      |            bracketed:
[L:  1, P: 20]      |                start_bracket:                                '('
[L:  1, P: 21]      |                newline:                                      '\n'
[L:  2, P:  1]      |                whitespace:                                   '    '
[L:  2, P:  5]      |                raw:                                          'username'
[L:  2, P: 13]      |                whitespace:                                   ' '
[L:  2, P: 14]      |                raw:                                          'TEXT'
[L:  2, P: 18]      |                comma:                                        ','
[L:  2, P: 19]      |                newline:                                      '\n'
[L:  3, P:  1]      |                whitespace:                                   '    '
[L:  3, P:  5]      |                raw:                                          'age'
[L:  3, P:  8]      |                whitespace:                                   ' '
[L:  3, P:  9]      |                raw:                                          'INT'
[L:  3, P: 12]      |                whitespace:                                   ' '
[L:  3, P: 13]      |                raw:                                          'CHECK'
[L:  3, P: 18]      |                bracketed:
[L:  3, P: 18]      |                    start_bracket:                            '('
[L:  3, P: 19]      |                    raw:                                      'age'
[L:  3, P: 22]      |                    whitespace:                               ' '
[L:  3, P: 23]      |                    raw:                                      '>'
[L:  3, P: 24]      |                    whitespace:                               ' '
[L:  3, P: 25]      |                    raw:                                      '18'
[L:  3, P: 27]      |                    end_bracket:                              ')'
[L:  3, P: 28]      |                newline:                                      '\n'
[L:  4, P:  1]      |                end_bracket:                                  ')'
[L:  4, P:  2]      |    statement_terminator:                                     ';'
[L:  4, P:  3]      |    newline:                                                  '\n'

==== parsing violations ====
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'CREATE TABLE users
                       | (\n username TEXT,\n...'
WARNING: Parsing errors found and dialect is set to 'ansi'. Have you configured your dialect?
```

## Dialect

ANSI

## Version

```console
$ sqlfluff --version
sqlfluff, version 0.6.6

$ python -V
Python 3.9.7
```

## Configuration

Standard configuration.

**Comment by tunetheweb** [comment]
The Postgres dialect supports this syntax but there probably is an argument to move it into ANSI as I think it's part of ANSI standard and quite a few other dialects support it AFAIK.

Which SQL are you using out of interest?

**Comment by tlgs** [comment]
Ah, you're absolutely correct.

I am using SQLite and unfortunately my use case is not a simple `greater than` check; I'm using the [`REGEXP`](https://www.sqlite.org/lang_expr.html#the_like_glob_regexp_and_match_operators) operator. In fact, I had previously tried `--dialect=postgres` and it failed.

I wrote the simple test script above and saw it failed with ANSI and so (wrongly) assumed that my previous run with Postgres
had failed due to the `CHECK` keyword.
It's now clear that it was because there's no REGEXP operator/function in Postgres (<https://www.postgresql.org/docs/current/functions-matching.html>).

This means I'd still have issues linting unless SQLite was supported as a dialect. :wink:

**Comment by tunetheweb** [comment]
Well I think we should add this to ANSI anyway. Just about to submit a PR for that.

What's your REGEXP query and will check if it now parses with this change?

**Comment by tlgs** [comment]
Here you go:

```sql
CREATE TABLE colors (
    css_name TEXT,
    rgb TEXT CHECK(rgb REGEXP '^#[0-9A-F]{6}$')
);
```

**Comment by tunetheweb** [comment]
This works:

```sql
CREATE TABLE colors (
    css_name TEXT,
    rgb TEXT CHECK(REGEXP '^#[0-9A-F]{6}$')
);
```

Or this:

```sql
CREATE TABLE colors (
    css_name TEXT,
    rgb TEXT CHECK(rgb LIKE REGEXP '^#[0-9A-F]{6}$')
);
```

But not the one you gave. That definitely correct?

**Comment by tlgs** [comment]
Yeah, just double checked. Neither of those work for SQLite.

**Comment by tunetheweb** [comment]
OK can you open another issue for SQLLite support? And specifically mention it's mostly like Postgres but this is one difference?

Leave this issue as think worth resolving the original ask.
